alacritty_terminal/src/term/mod.rs

```diff
@@ -1256,10 +1268,55 @@ impl<T> Term<T> {
         self.event_proxy.send_event(Event::Exit);
     }
 
+    #[inline]
     pub fn clipboard(&mut self) -> &mut Clipboard {
         &mut self.clipboard
     }
 
+    /// Toggle the keyboard motion mode.
+    #[inline]
+    pub fn toggle_keyboard_mode(&mut self) {
+        self.mode ^= TermMode::KEYBOARD_MOTION;
+        self.grid.selection = None;
+
+        // Reset keyboard cursor position to match primary cursor
+        if self.mode.contains(TermMode::KEYBOARD_MOTION) {
+            let line = min(self.cursor.point.line + self.grid.display_offset(), self.lines() - 1);
```

Comment:
Could you explain to me this computations? Like I do understand the result, but I don't get your intention here. I feel like if your `display_offset` isn't 0, you should scroll to your normal cursor?

Reply:
The normal cursor should never, ever move.

All we do here is we set the position of the keyboard cursor to the position of the normal cursor. If the normal cursor is not on screen, we instead use the bottom line instead.

Reply:
I was about using `ScrollToBottom` when you leave keyboard mode, so you'll go back to your normal cursor position. Like now you should press something to trigger it. This is what tmux is doing.

Reply:
Ah, that's actually a good point. Yeah, makes sense.

Reply:
Actually I think I'm against implementing this directly in code and would instead like to make this user-configurable.

I'm personally not sure which I prefer for example, but if we don't implement it directly in Alacritty, everyone can choose by themselves simply by adding a keybind to scroll to the bottom when leaving the keyboard motion mode. So the question is only what the default should be.

I take it @kchibisov you'd vote to make it the default to scroll automatically? @nixpulvis what do you think?

One important point is that it's probably easier to add it when it is not the default, rather than remove it when it is, but both should be simple to change in the config.

Reply:
I've chosen to go with going back to the bottom for now. The idea behind that was that the cursor will always be visible like that, unless explicitly scrolled with the mouse.

I'll have to try it out for a bit to see how I like it though. Still let me know what you think @nixpulvis.

Reply:
@chrisduerr I think I'd expect the cursor to return to exactly wherever it was before the mode was activated. I believe that is what will be accomplished by the current implementation, since I'm unaware of a way to set the cursor's position such that the terminal thinks the end is past the cursor, but maybe I'm forgetting an escape or something.

Reply:
> I believe that is what will be accomplished by the current implementation

It is not. The current implementation always jumps back to the bottom.